### PR TITLE
feat(core): add flag to set dependant libs in either peerDependencies or dependencies

### PR DESCRIPTION
**Pull request by bhuber2010**
## Current Behavior (This is the behavior we have today, before the PR is merged)
When building libraries, library's package.json dependencies is updated with any node_module deps.

## Expected Behavior (This is the new behavior we can expect after the PR is merged)
When building libraries, library's package.json peerDependencies is updated with any node_module deps.
## Issue
Addresses: https://github.com/nrwl/nx/pull/2879#issuecomment-632406767

### CODE REVIEW

File: packages/workspace/src/utils/buildable-libs-utils.ts [hunk 1]
```diff
@@ -280,7 +280,7 @@ export function updateBuildableProjectPackageJsonDependencies(
           );
           depVersion = readJsonFile(depPackageJsonPath).version;
 
-          packageJson.dependencies[packageName] = depVersion;
+          packageJson.peerDependencies[packageName] = depVersion;
```
Comment (Cammisuli) [inline packages/workspace/src/utils/buildable-libs-utils.ts:283]: I think that Nx libraries should still be considered as dependencies. Maybe you should make the npm section below add to peerDependencies instead.

Comment (vmcbaptista) [reply packages/workspace/src/utils/buildable-libs-utils.ts:283]: @Cammisuli this PR is based in a comment I did in #2879.
I actually was thinking that both the nx and npm dependencies would end up as peerDependencies.
Having nx dependencies as dependencies can cause dependencies trees like this, which I believe is not a desirable situation
```
root
+-- @my-org/foo@1
+-- @my-org/bar@1
|   +-- @my-org/foo@2
```

@bhuber2010 I see that already a month passed and you did not do any changes. Do you plan to continue working on this? I can take over if you are currently busy with other stuff.

Comment (bhuber2010) [reply packages/workspace/src/utils/buildable-libs-utils.ts:283]: O wow, totally missed the notification I guess, sorry guys! And to myself, was wondering about this. And yea I agree on both going to peerDeps. Didn't realize I missed that, will do that right now and update this PR.

Comment (bhuber2010) [reply packages/workspace/src/utils/buildable-libs-utils.ts:289]: updated

**Comment by Cammisuli** [comment]
Sorry folks, can we rebase this branch and fix the merge conflicts?

**Comment by bhuber2010** [comment]
I'll try and get this updated today.

**Review by Cammisuli — APPROVED** [review]

**Comment by bhuber2010** [comment]
I don't really know what's going on with e2e 5 and 7. But it'd be good to get this merged in soon.

**Comment by Cammisuli** [comment]
the e2e 5 is stuff to do with react, and those are fixed now. But for e2e 7, it's something that has to do with node, which this change touched. So it's something that needs to be investigated. I'll have to slice out some time to take a look at the e2e and make sure everything is good. Sorry about that ☹️

**Comment by Cammisuli** [comment]
Here's an update for this PR. We're going ahead with the proposed changed, but I'm going to be changing it so that UI libraries (React and Angular) or libraries that depend on a framework (Nestjs) will set the deps as `peerDependencies`. Everything else (specifically just pure Node libraries) will set as `dependencies`. 

I'm going to make the change so that the function that adds the dependants would take an option to specify what kind of dependencies it needs to add to.

**Review by FrozenPandaz — APPROVED** [review]
LGTM!

**Comment by bhuber2010** [comment]
awesome, thanks guys!

**Pull request merged**